app/models/user.rb

```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+
+  def tests_by_level(level)
+    Test
+      .joins("JOIN user_tests ON tests.id = user_tests.test_id")
+      .where("user_id = :id AND level = :level", id: id, level: level)
```

Comment:
```suggestion
      .where(user_id: id, level: level)
```
так лучше
ну или можно еще таблицы указать, как в модели Тест было (см. выше)